hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/OMDBInsightEndpoint.java

```diff
@@ -0,0 +1,412 @@
+        pendingForDeletionKeyInfo)).build();
+  }
+
+  private KeyInsightInfoResp getPendingForDeletionDirInfo(
+      int limit, String prevKeyPrefix,
+      KeyInsightInfoResp pendingForDeletionKeyInfo) {
+
+    List<KeyEntityInfo> deletedDirInfoList =
+        pendingForDeletionKeyInfo.getDeletedDirInfoList();
+
+    Table<String, OmKeyInfo> deletedDirTable =
+        omMetadataManager.getDeletedDirTable();
+    try (
+        TableIterator<String, ? extends Table.KeyValue<String, OmKeyInfo>>
+            keyIter = deletedDirTable.iterator()) {
+      boolean skipPrevKey = false;
+      String seekKey = prevKeyPrefix;
+      if (StringUtils.isNotBlank(prevKeyPrefix)) {
+        skipPrevKey = true;
+        Table.KeyValue<String, OmKeyInfo> seekKeyValue =
+            keyIter.seek(seekKey);
+        // check if RocksDB was able to seek correctly to the given key prefix
+        // if not, then return empty result
+        // In case of an empty prevKeyPrefix, all the keys are returned
+        if (seekKeyValue == null ||
+            (StringUtils.isNotBlank(prevKeyPrefix) &&
+                !seekKeyValue.getKey().equals(prevKeyPrefix))) {
+          return pendingForDeletionKeyInfo;
+        }
+      }
+      while (keyIter.hasNext()) {
+        Table.KeyValue<String, OmKeyInfo> kv = keyIter.next();
+        String key = kv.getKey();
+        OmKeyInfo omKeyInfo = kv.getValue();
+        // skip the prev key if prev key is present
+        if (skipPrevKey && key.equals(prevKeyPrefix)) {
+          continue;
+        }
+        KeyEntityInfo keyEntityInfo = new KeyEntityInfo();
+        keyEntityInfo.setKey(key);
+        keyEntityInfo.setPath(omKeyInfo.getKeyName());
+        keyEntityInfo.setInStateSince(omKeyInfo.getCreationTime());
+        keyEntityInfo.setSize(omKeyInfo.getDataSize());
+        keyEntityInfo.setReplicatedSize(omKeyInfo.getReplicatedSize());
+        keyEntityInfo.setReplicationConfig(omKeyInfo.getReplicationConfig());
+        pendingForDeletionKeyInfo.setUnreplicatedTotal(
+            pendingForDeletionKeyInfo.getUnreplicatedTotal() +
+                keyEntityInfo.getSize());
+        pendingForDeletionKeyInfo.setReplicatedTotal(
+            pendingForDeletionKeyInfo.getReplicatedTotal() +
+                keyEntityInfo.getReplicatedSize());
+        deletedDirInfoList.add(keyEntityInfo);
+        if (deletedDirInfoList.size() == limit) {
+          break;
+        }
+      }
+    } catch (IOException ex) {
+      throw new WebApplicationException(ex,
+          Response.Status.INTERNAL_SERVER_ERROR);
+    } catch (IllegalArgumentException e) {
+      throw new WebApplicationException(e, Response.Status.BAD_REQUEST);
+    } catch (Exception ex) {
+      throw new WebApplicationException(ex,
+          Response.Status.INTERNAL_SERVER_ERROR);
+    }
+    return pendingForDeletionKeyInfo;
+  }
+
+  private KeyInsightInfoResp getPendingForDeletionKeyInfo(
```

Comment:
Return KeyInsightInfoResp , Same object passed as input param KeyInsightInfoResp is returned post modification. Return seems could be void.

Reply:
@krishnaasawa1 thanks for review, however this returned is not  void, this KeyInsightInfoResp is passed as a empty object to populate pendingForDeltionKeys to getPendingForDeletionKeyInfo method and later after populating it , it is being passed as reference to "getPendingForDeletionDirInfo" method.

Reply:
What I meant as object is passed  after creating new, no need to return back KeyInsightInfoResp  as calling method already have reference of same object as it only has passed as input param . It's okay to keep , if required.

Reply:
@krishnaasawa1  - Thanks for elaborating, I understood your point. Will make a change.

Reply:
@krishnaasawa1 - Pls re-review. Changes pushed for making methods return as void.